package/test/src/index.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  DetectVersionFilesOutput,
+  DownloadPrebuiltInput,
+  DownloadPrebuiltOutput,
+  LoadVersionsInput,
+  LoadVersionsOutput,
+  LocateExecutablesInput,
+  LocateExecutablesOutput,
+  ParseVersionFileInput,
+  ParseVersionFileOutput,
+  RegisterToolInput,
+  RegisterToolOutput,
+  ResolveVersionInput,
+  ResolveVersionOutput,
+  UnresolvedVersionSpec,
+  VerifyChecksumInput,
+  VerifyChecksumOutput,
+  VersionLike,
+  VersionSpec,
+  createDetectVersionFiles,
+  createDownloadPrebuilt,
+  createLoadVersions,
+  createLocateExecutables,
+  createParseVersionFile,
+  createRegisterTool,
+  createResolveVersion,
+  createVerifyChecksum,
+  fetchUrl,
+  getHostEnvironment,
+  getToolConfig,
+  hostLog,
+} from "@moonrepo/proto-pdk";
+
+export const register_tool = createRegisterTool(
+  (input: RegisterToolInput): RegisterToolOutput => {
+    // TODO: hostLog is a bit awkward to use
```

Comment:
I reckon something like `hostLog.trace('Hello world', optionalDataObject)` would be nicer. Alternatively could just be overloads, but that truly sucks to implement in javascript/typescript.